user-interface/src/lib/components/combobox/ComboBox.test.tsx

```diff
@@ -0,0 +1,692 @@
+import { LegacyRef } from 'react';
+import ComboBox, { ComboBoxProps, ComboOption } from './ComboBox';
+import { ComboBoxRef } from '@/lib/type-declarations/input-fields';
+import { fireEvent, render, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+
+const comboboxId = 'test-combobox';
+
+function toggleDropdown(id: string) {
+  const toggleButton = document.querySelector(`#${id}-expand`);
+  fireEvent.click(toggleButton!);
+  const dropdownList = document.querySelector('ul');
+  return dropdownList;
+}
+
+function focusComboInputField(id: string) {
+  const inputField = document.querySelector(`#${id}-combo-box-input`);
+  fireEvent.click(inputField!);
+  return inputField;
```

Comment:
This return seems like a side effect. Can we split getting the input field from giving it focus?